src/contexts/states/viewStore.ts

```diff
@@ -0,0 +1,245 @@
+} from "../../typings/worker";
+import {
+    ACTION_NAME,
+    NavigationAction,
+} from "../../utils/actions";
+import {clamp} from "../../utils/math";
+import {updateWindowUrlHashParams} from "../UrlContextProvider";
+import useContextStore from "./contextStore";
+import useLogFileManagerStore from "./LogFileManagerProxyStore";
+import useLogFileStore from "./logFileStore";
+import useQueryStore from "./queryStore";
+import useUiStore from "./uiStore";
+
+
+interface ViewStoreValues {
+    beginLineNumToLogEventNum: BeginLineNumToLogEventNumMap;
+    isPrettified: boolean;
+    logData: string;
+    numPages: number;
+    pageNum: number;
+}
+
+interface ViewStoreActions {
+    setBeginLineNumToLogEventNum: (newMap: BeginLineNumToLogEventNumMap) => void;
+    setIsPrettified: (newIsPrettified: boolean) => void;
+    setLogData: (newLogData: string) => void;
+    setNumPages: (newNumPages: number) => void;
+    setPageNum: (newPageNum: number) => void;
+
+    updatePageData: (pageData: PageData) => void;
+    loadPageByAction: (navAction: NavigationAction) => void;
+    filterLogs: (filter: LogLevelFilter) => void;
+}
+
+type ViewState = ViewStoreValues & ViewStoreActions;
+
+const VIEW_STORE_DEFAULT: ViewStoreValues = {
+    beginLineNumToLogEventNum: new Map<number, number>(),
+    isPrettified: false,
+    logData: "No file is open.",
+    numPages: 0,
+    pageNum: 0,
+};
+
+/**
+ * Returns a `PAGE_NUM` cursor based on a navigation action.
+ *
+ * @param navAction Action to navigate to a new page.
+ * @param currentPageNum
+ * @param numPages
+ * @return `PAGE_NUM` cursor.
+ */
+const getPageNumCursor = (
+    navAction: NavigationAction,
+    currentPageNum: number,
+    numPages: number
+): Nullable<CursorType> => {
+    let newPageNum: number;
+    let position: EVENT_POSITION_ON_PAGE;
+    switch (navAction.code) {
+        case ACTION_NAME.SPECIFIC_PAGE:
+            position = EVENT_POSITION_ON_PAGE.TOP;
+
+            // Clamp is to prevent someone from requesting non-existent page.
+            newPageNum = clamp(navAction.args.pageNum, 1, numPages);
+            break;
+        case ACTION_NAME.FIRST_PAGE:
+            position = EVENT_POSITION_ON_PAGE.TOP;
+            newPageNum = 1;
+            break;
+        case ACTION_NAME.PREV_PAGE:
+            position = EVENT_POSITION_ON_PAGE.BOTTOM;
+            newPageNum = clamp(currentPageNum - 1, 1, numPages);
+            break;
+        case ACTION_NAME.NEXT_PAGE:
+            position = EVENT_POSITION_ON_PAGE.TOP;
+            newPageNum = clamp(currentPageNum + 1, 1, numPages);
+            break;
+        case ACTION_NAME.LAST_PAGE:
+            position = EVENT_POSITION_ON_PAGE.BOTTOM;
+            newPageNum = numPages;
+            break;
+        default:
+            return null;
+    }
+
+    return {
+        code: CURSOR_CODE.PAGE_NUM,
+        args: {pageNum: newPageNum, eventPositionOnPage: position},
+    };
+};
+
+// eslint-disable-next-line max-lines-per-function
```

Comment:
as a follow up to a previous discussion about suppressing this only for Zustand store create function calls:

It seems there's no specific way to supress the check with any regex matches like some other rules. We can add the `eslint-disable-next-line` directive for Zustand store create calls as needed.